Non greedy robustness

rimashahbazyan opened the pull request.

rimashahbazyan committed:
score readme added

rimashahbazyan committed:
generate until task's "until" parameter's default value fixed.

rimashahbazyan committed:
score mmlu-pro and agieval added

rimashahbazyan committed:
changed macro accuracy to micro for agieval

rimashahbazyan committed:
Always E removed from agi eval

rimashahbazyan committed:
redundancies removed

rimashahbazyan committed:
MATH added

rimashahbazyan committed:
minor cosmetic changes for math

rimashahbazyan committed:
Licenses added Readme updated

rimashahbazyan committed:
changes for flake8 + license header on math

rimashahbazyan committed:
Score added to readme and precommit was run.

rimashahbazyan committed:
Score added to readme and precommit was run.

rimashahbazyan committed:
Merge branch 'score_tasks' of github.com:rimashahbazyan/lm-evaluation-harness into score_tasks

rimashahbazyan committed:
Import error fixed

rimashahbazyan committed:
math task bugfix
postprocess minor fix

CR for math added

rimashahbazyan committed:
math task bugfix
postprocess minor fix

rimashahbazyan committed:
CR for math added

rimashahbazyan committed:
math CR

rimashahbazyan committed:
Merge branch 'score_tasks' of github.com:rimashahbazyan/lm-evaluation-harness into score_tasks

rimashahbazyan committed:
Math cr fixed

rimashahbazyan committed:
mmlu_pro non_greedy task added

rimashahbazyan committed:
non greedy summarizer added

rimashahbazyan committed:
Non greedy for all score tasks

rimashahbazyan committed:
Bugfixes for non-greedy

rimashahbazyan committed:
Merge branch 'main' of github.com:rimashahbazyan/lm-evaluation-harness into non_greedy_robustness

rimashahbazyan committed:
fixing the until argument

rimashahbazyan committed:
undoing the change to "until" arguments default behaviour

rimashahbazyan committed:
minor fix in summarizer

rimashahbazyan committed:
log naming changes for better readability

rimashahbazyan committed:
math subtasks naming fix

rimashahbazyan committed:
agieval subtask naming fix

rimashahbazyan committed:
logging added for debugging

rimashahbazyan committed:
path issue fixed

rimashahbazyan committed:
minor fix

rimashahbazyan committed:
path fix

rimashahbazyan committed:
path fix

rimashahbazyan committed:
non_greedy_math minor fix

rimashahbazyan committed:
final changes

ekmb commented on lm_eval/tasks/score/NON_GREEDY.md:
```diff
@@ -0,0 +1,32 @@
+   * fixing the seed with the run argument `--seed=1`
+   * passing the `--log_samples` argument
+   * specifying an output with `--output_path=SOME_OUTPUT_PATH/seed_1`
+   * if running with vllm it is important to set the seed in the `--model_args` as well buy jsut specifying the `seed` parameter
```
"buy jsut" -> "by just"

ekmb replied:
Any caution message here that prediction will take some space? and why file saving is required

ekmb commented on lm_eval/tasks/score/NON_GREEDY.md:
```diff
@@ -0,0 +1,32 @@
+   * specifying an output with `--output_path=SOME_OUTPUT_PATH/seed_1`
+   * if running with vllm it is important to set the seed in the `--model_args` as well buy jsut specifying the `seed` parameter
+
+2. Repeat the process for another 4 times by changing the seed to 2, 3, 4 and 5. Equivalantly adjusting the `--output_path` argument so that the directory is named as `seed_{SEED VALUE FOR THE RUN}`
```
typo in "Equivalantly" but how about remove this word...
"Repeat the process 4 times by changing the `--seed` and `--output_path` accordingly"

ekmb commented on lm_eval/tasks/score/NON_GREEDY.md:
`````diff
@@ -0,0 +1,32 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+````
`````
add more details on the non-greedy eval? 5 seeds various tasks, fixed prompt, supported datasets? expected result? i.e. we don't want variance based on the seed alone
temp or other inference params that are worth mentioning?

ekmb replied:
you have a high level overview of the task but I'd repeat it here as some details are still missing or expand the one in the README and put a reference here.

ekmb commented on lm_eval/tasks/score/non_greedy_summarizer.py:
```diff
@@ -0,0 +1,284 @@
+import argparse
```
nvidia header is missing

rimashahbazyan committed:
changed readme for non-greedy
added Nvidia header
added wxample script for non_greedy
changed prompts to match that fo trt runs

rimashahbazyan committed:
non greedy summarizer bugfix

rimashahbazyan committed:
non_greedy summarizer fixed
